lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -434,36 +434,51 @@ def draw(self, renderer):
 
         # add the projection matrix to the renderer
         self.M = self.get_proj()
-        renderer.M = self.M
-        renderer.vvec = self.vvec
-        renderer.eye = self.eye
-        renderer.get_axis_position = self.get_axis_position
-
-        # Calculate projection of collections and patches and zorder them.
-        # Make sure they are drawn above the grids.
-        zorder_offset = max(axis.get_zorder()
-                            for axis in self._get_axis_list()) + 1
-        for i, col in enumerate(
-                sorted(self.collections,
-                       key=lambda col: col.do_3d_projection(renderer),
-                       reverse=True)):
-            col.zorder = zorder_offset + i
-        for i, patch in enumerate(
-                sorted(self.patches,
-                       key=lambda patch: patch.do_3d_projection(renderer),
-                       reverse=True)):
-            patch.zorder = zorder_offset + i
-
-        if self._axis3don:
-            # Draw panes first
-            for axis in self._get_axis_list():
-                axis.draw_pane(renderer)
-            # Then axes
-            for axis in self._get_axis_list():
-                axis.draw(renderer)
+        props3d = {
+            name: cbook.deprecated('3.4', name=name,
+                                   alternative=f'self.axes.{name}')(
+                property(lambda self, _value=getattr(self, name): _value))
+            for name in ['M', 'vvec', 'eye', 'get_axis_position']
+        }
 
-        # Then rest
-        super().draw(renderer)
+        with cbook._setattr_cm(type(renderer), **props3d):
+            def do_3d_projection(artist):
+                if artist.__module__ == 'mpl_toolkits.mplot3d.art3d':
+                    # Our 3D Artists have deprecated the renderer parameter.
+                    return artist.do_3d_projection()
```

Comment:
No, not when we're calling it ourselves directly.

Reply:
Sorry I haven't replied back to this in awhile.

What I mean is, if our Artists are triggering the deprecation warning, doesn't that mean that we missed something here?

Reply:
There are two deprecations: 1) Accessing the attributes on the `renderer` and 2) passing a non-*None* `renderer` to `do_3d_projection`. I think you're talking about the former, but this is about avoiding the latter.

Reply:
No, I was talking about the latter. If we have code within our codebase that passes a non-None renderer, we should want to know about that. I can easily envision a situation where after this PR is merged another PR that was started before now accidentally adds code that passes an argument back in. We would want to know about that, right?

Reply:
_This_ is the place that passes the renderer, and now it doesn't, which is what we want so it doesn't trigger the deprecation. I'm sorry, but I still don't understand your question.

Reply:
@WeatherGod My understanding here is the whole reason for this block is to avoid passing `renderer` when we're working with our own artists. So this is not swallowing the warning, it's taking special effort to avoid triggering it, using the behavior we want to be normal in the future. For current compatibility with external artists, there's still a path that passes `renderer`.

Reply:
So, my question still remains unaddressed. What's going to happen if a future MR accidentally passes `renderer`? From what I see here, it will completely mask it, right?

Reply:
And just to note, I am not going to block this if two other developers feel that this is fine. I just feel that I haven't gotten a satisfactory answer to this question. Maybe the answer is that "we'll be able to fix them when we remove this shim and tests fail", or maybe someone can explain how this can't happen, but so far, I haven't seen either.

Reply:
Passes it _where_? In any case, all our artists will trigger a deprecation.

Reply:
OMG, I think I finally realize the mistake in my understanding. I am *so* sorry for being so obtuse about this. I had honestly thought that this was some sort of extremely weird way of wrapping the `do_3d_projection()` method for all of the artist objects, and so any uses of `do_3d_projection(renderer)` anyplace else in this codebase or in a user's script would fail to trigger the warning.

So, if I am reading this correctly now, this wrapping should be strictly contained within just the 3d draw stack.

If so, then this all absolutely makes sense.

Reply:
Ah. Yes, this is a nested function that can only be seen within `draw`.